### PR TITLE
Feature/germantranslation

### PR DESCRIPTION
**Pull request by EliasKotlyar**
Wanted to make #694 work. Refactored the translation System to CSV-Files. Added one test-translation for German Language.

### CODE REVIEW

**Comment by pkarw** [comment]
Nice job! I’ve added quite a few translations today as I was fixing bugs before 1.0RC release

Could you just merge the new labels to CSV?

**Comment by pkarw** [comment]
Oh, and somebody just need to provide us with real DE translations :)

File: core/lib/themes.js [hunk 1]
```diff
@@ -8,10 +7,37 @@ import * as themeMixins from 'theme/mixins'
...
+}
+
 /**
  * Return messages load from both: core/resource/i18n and theme/resource/i18n
  */
 export function translations () {
+  const coreMessages = {
+    'en-US':
+      convertToObject(require('dsv-loader?rows!core/resource/i18n/en-US.csv')),
+    'de-DE':
```
Comment (pkarw) [inline core/lib/themes.js]: I’m just wondering about the optimization: can we add the conversion from CSV to JSON as a build step (you may just extend package.js ‘build’ and ‘dev’ scripts to another command); the command itself should be placed in core/scripts and used
To generate the /dist/i18n.js

I’m not sure about processing the /theme/*/resource/i18n.json -
Is your solution compliant with this feature of theme related translations?

**Comment by pkarw** [comment]
By theme related translations I mean: https://github.com/DivanteLtd/vue-storefront/blob/master/core/lib/themes.js

It will be really cool to have CSV support but it mus support theme related translations + it must not parse CSV at the runtime because of performance

**Comment by EliasKotlyar** [comment]
I have commited a new version. Its now running with the webpack, and compiling both theme-translations and normal ones. Can you please devtest it?

**Comment by pkarw** [comment]
... and please remove i18n.json from themes

After this change I will be able to merge this in :)

**Comment by pkarw** [comment]
Oh, and please update the docs: https://github.com/DivanteLtd/vue-storefront/blob/master/doc/i18n/Working%20with%20translations.md

Sorry for bothering you with so many tasks but it's quite complex topic - the internationalization :)

**Comment by pkarw** [comment]
@EliasKotlyar you did great job here. Maybe you join our Slack and maybe are you willing to take some more tasks from Issues?

**Comment by EliasKotlyar** [comment]
Hello pkarw,

Thanks you for inviting me. I already joined Slack.

Greatings
Elias

**Pull request merged**